apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -143,6 +142,136 @@ handle_request('PostOracleUnsubscribe', _Req, _Context) ->
     {501, [], #{}};
 
+handle_request('PostNamePreclaimTx', #{'NamePreclaimTx' := NamePreclaimTxObj}, _Context) ->
+    case get_local_pubkey_with_next_nonce() of
+        {ok, PubKey, Nonce} ->
+            #{<<"commitment">> := Commitment,
+              <<"fee">>        := Fee} = NamePreclaimTxObj,
+            case aec_base58c:safe_decode(commitment, Commitment) of
+                {ok, DecodedCommitment} ->
+                    {ok, PreclaimTx} =
+                        aens_preclaim_tx:new(
+                          #{account    => PubKey,
+                            nonce      => Nonce,
+                            commitment => DecodedCommitment,
+                            fee        => Fee}),
+                    sign_and_push_to_mempool(PreclaimTx),
+                    {200, [], #{commitment => Commitment}};
+                {error, _Reason} ->
+                    {400, [], #{reason => <<"Invalid commitment hash">>}}
+            end;
+        {error, account_not_found} ->
+            {404, [], #{reason => <<"No funds in an account">>}};
+        {error, key_not_found} ->
+            {400, [], #{reason => <<"Keys not configured">>}}
+    end;
+
+handle_request('PostNameClaimTx', #{'NameClaimTx' := NameClaimTxObj}, _Context) ->
+    case get_local_pubkey_with_next_nonce() of
+        {ok, PubKey, Nonce} ->
+            #{<<"name">>       := Name,
+              <<"name_salt">> := NameSalt,
+              <<"fee">>        := Fee} = NameClaimTxObj,
+            {ok, ClaimTx} =
+                aens_claim_tx:new(
+                  #{account    => PubKey,
+                    nonce      => Nonce,
+                    name       => Name,
+                    name_salt => NameSalt,
+                    fee        => Fee}),
+            sign_and_push_to_mempool(ClaimTx),
+            {200, [], #{name_hash => aec_base58c:encode(name, aens_hash:name_hash(Name))}};
+        {error, account_not_found} ->
+            {404, [], #{reason => <<"No funds in an account">>}};
+        {error, key_not_found} ->
+            {400, [], #{reason => <<"Keys not configured">>}}
+    end;
+
+handle_request('PostNameUpdateTx', #{'NameUpdateTx' := NameUpdateTxObj}, _Context) ->
+    case get_local_pubkey_with_next_nonce() of
+        {ok, PubKey, Nonce} ->
+            #{<<"name_hash">> := NameHash,
+              <<"name_ttl">>  := NameTTL,
+              <<"pointers">>  := Pointers,
+              <<"ttl">>       := TTL,
+              <<"fee">>       := Fee} = NameUpdateTxObj,
+            case aec_base58c:safe_decode(name, NameHash) of
+                {ok, DecodedNameHash} ->
+                    {ok, UpdateTx} =
+                        aens_update_tx:new(
+                          #{account   => PubKey,
+                            nonce     => Nonce,
+                            name_hash => DecodedNameHash,
+                            name_ttl  => NameTTL,
+                            pointers  => jsx:decode(Pointers,[{labels, atom}]),
```

Comment:
I understand this is an internal API, though did you consider not creating atoms?

Reply:
@lucafavatella I went with atoms, to match atoms in internal in the dispatcher. It is an injection verctor, true. We should refactor.